presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -135,15 +130,10 @@ private void unspillNextLookupSource()
         checkState(spilledPartitions.isPresent());
         if (spilledPartitions.get().hasNext()) {
             int currentSpilledPartition = spilledPartitions.get().next();
-
-            spilledLookupJoiner = Optional.of(new SpillledLookupJoiner(
-                    allTypes,
-                    lookupSourceFactory.readSpilledLookupSource(operatorContext.getSession(), currentSpilledPartition),
-                    joinProbeFactory,
-                    spiller.get().getSpilledPages(currentSpilledPartition),
-                    probeOnOuterSide));
+            currentLookupPartition = Optional.of(lookupSourceFactory.unspillLookupPartition(currentSpilledPartition));
```

Comment:
Maybe add some `checkState` before overwriting that previous partition has been fully processed? Maybe that will catch some bug?